lib/cartopy/tests/mpl/conftest.py

```diff
@@ -20,6 +20,7 @@ def mpl_test_cleanup(request):
         plt.switch_backend('Agg')
         stack.enter_context(plt.rc_context())
         yield
+        plt.close('all')
```

Comment:
You should make this a `stack.callback` before the backend switch so it runs on context exit.